test/main/utilities.js

```diff
@@ -0,0 +1,43 @@
+QUnit.module( "QUnit.objectType" );
+
+function buildFunctions() {
+	var objects = [ function( ) { } ];
+
+	try {
+		var arrow = new Function( "return () => {};" )( );
+		objects.push( arrow );
+	} catch ( e ) { }
+
+	try {
+		var async = new Function( "return async function( ) { };" )( );
+		objects.push( async );
+	} catch ( e ) { }
```

Comment:
I got it! Thanks for the clarification! 

btw eval allows a shorter syntax, but does the same:
`new Function( "return async function( ) { };" )( )` => `eval( "async function( ) {};" )`